### PR TITLE
Adding AsyncAws adapter

### LINKED ISSUE #1220: AsyncAws adapter
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| BC Break    | no

#### Summary

For the past few months I've been maintaining an adapter for AsyncAws S3. I've been following the development since beta 1. I notice that it is very similar to the AWS S3 adapter. 

In a very humble request, should I add the AsyncAws S3 adapter under the Flysystem namespace?

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #1220

### CODE REVIEW

**Comment by Nyholm** [comment]
Some unit tests are failing, working on them now

Edit: fixed!

**Comment by Nyholm** [comment]
The FTP test seams to fail randomly.

File: src/AsyncAwsS3/AsyncAwsS3Adapter.php [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+use League\Flysystem\UnableToSetVisibility;
+use League\Flysystem\Visibility;
+use League\MimeTypeDetection\FinfoMimeTypeDetector;
+use League\MimeTypeDetection\MimeTypeDetector;
+use Throwable;
+
+class AsyncAwsS3Adapter implements FilesystemAdapter
+{
+    /**
+     * @var array
```
Comment (OskarStark) [inline src/AsyncAwsS3/AsyncAwsS3Adapter.php]:
```suggestion
     * @var string[]
```
?

File: src/AsyncAwsS3/AsyncAwsS3Adapter.php [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+        'SSECustomerAlgorithm',
+        'SSECustomerKey',
+        'SSECustomerKeyMD5',
+        'SSEKMSKeyId',
+        'ServerSideEncryption',
+        'StorageClass',
+        'Tagging',
+        'WebsiteRedirectLocation',
+    ];
+    private const EXTRA_METADATA_FIELDS = [
```
Comment (OskarStark) [inline src/AsyncAwsS3/AsyncAwsS3Adapter.php:66]:
```suggestion

    /**
     * @var string[]
     */
    private const EXTRA_METADATA_FIELDS = [
```

**Comment by Nyholm** [comment]
Finally all done and tests are green

**Comment by Nyholm** [comment]
The PR is rebased. 

I've not added any changes in the subsplit command.

**Review by jderusse — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you Oskar for your suggestions. I updated the AwsS3V3 adapter with the same changes.

**Comment by Nyholm** [comment]
Thank you for merging.

**Comment by frankdejonge** [comment]
@Nyholm thank you for contributing! I'm tackling the flacky behavior of the FTP failures. It'll be a trade-off, but at least it'll be more stable. So you won't have to deal with that anymore in upcoming PR's.

**Pull request merged**